packages/compiler/src/compiler/base/nodes/tags/message.ts

```diff
@@ -107,9 +109,10 @@ function buildMessage<R extends MessageRole>(
 
   if (role === MessageRole.assistant) {
     return {
+      ...attributes,
       role,
       toolCalls: toolCalls.map(({ value }) => value),
-      content: (content[0]! as TextContent).text,
+      content,
```

Comment:
so has content been transformed before? I don't see where we are getting the content from the array now